astropy/units/function/magnitude_zero_points.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+"""
+This package defines magnitude zero points.  By default, they are used to
+define corresponding magnitudes, but not enabled as regular physical units.
+To enable them, do::
+
+    >>> from astropy.units import magnitude_zero_points
+    >>> magnitude_zero_points.enable()  # doctest: +SKIP
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import numpy as _numpy
+from ..core import UnitBase, def_unit
+
+from ...constants import si as _si
+from .. import si, astrophys
+
+
+_ns = globals()
+
+def_unit(['Bol', 'L_bol'], _si.L_bol0, namespace=_ns, prefixes=False,
+         doc="Luminosity corresponding to absolute bolometric magnitude zero")
+def_unit(['bol', 'f_bol'], L_bol / (4 * _numpy.pi * (10.*astrophys.pc)**2),
+         namespace=_ns, prefixes=False, doc="Irradiance corresponding to "
+         "appparent bolometric magnitude zero")
+def_unit(['AB'], 10.**(-0.4*48.6) * 1.e-3 * si.W / si.m**2 / si.Hz,
+         namespace=_ns, prefixes=False,
+         doc="AB magnitude zero flux density.")
+def_unit(['ST'], 10.**(-0.4*21.1) * 1.e-3 * si.W / si.m**2 / si.AA,
```

Comment:
The main reason I did this originally (it was in the function unit code without the whole `def_unit` part) is that then the magnitude representation looks much nicer: `mag(ST)`, etc., without any need to fiddle with the `repr`. 

But this definitely is one of the reasons I didn't want to expose these zero points by default: they are really meant for magnitudes, not as regular units.